src/getJsonPathForNode.ts

```diff
@@ -0,0 +1,34 @@
+import { JsonPath } from '@stoplight/types';
+import * as Unist from 'unist';
+
+export const getJsonPathForNode = (root: Unist.Parent, node: Unist.Node): JsonPath | void => {
+  const path: JsonPath = [];
+  findNode(root, node, path);
+  return path;
+};
+
+function findNode(root: Unist.Parent | Unist.Node, node: Unist.Node, path: JsonPath): Unist.Node | undefined {
+  if (
+    node.position!.start.line === root.position!.start.line &&
+    node.position!.end.line === root.position!.end.line &&
+    node.position!.start.column === root.position!.start.column &&
+    node.position!.end.column === root.position!.end.column
```

Comment:
Re-work to check for node.position and root.position higher up, and remove `!` clobbers.